src/core/elf.c

```diff
@@ -15,22 +59,73 @@ elf_header_t* elf_load(uint64_t* data) {
     }
 
     DEBUG(
-        "file header: machine=%#x version=%#x entry=%p header_size=%u "
-        "ph_size=%u ph_num=%d ph_offset=%u sh_num=%u sh_num=%d",
-        elf->machine, elf->version, elf->entry, elf->header_size, elf->ph_size,
-        elf->ph_num, elf->ph_offset, elf->sh_size, elf->sh_num
-    );
+        "file header: machine=%#x version=%#x type=%d entry=%p header_size=%u "
+        "ph_size=%u ph_num=%d ph_offset=%llu sh_size=%u sh_num=%d sh_offset=%llu "
+        "strtab_index=%d",
+        elf->machine, elf->version, elf->type, elf->entry, elf->header_size,
+        elf->ph_size, elf->ph_num, elf->ph_offset, elf->sh_size, elf->sh_num,
+        elf->sh_offset, elf->strtab_index);
+
+    elf_section_header_t* sections = elf_section_header(elf);
+
+    //We need two phases to load all the sections before loading the program code. For now we only do one.
```

Comment:
Nit: My "brain linter" tells me that we should have a space after `//` 😉

Reply:
Done